addons/account_facturx/models/account_move.py

```diff
@@ -234,8 +234,8 @@ def message_post(self, **kwargs):
             self.env.context.get('default_type', self.type) in self.env['account.move'].get_invoice_types(include_receipts=True)
             or self.env['account.journal'].browse(self.env.context.get('default_journal_id')).type in ('sale', 'purchase')
         ):
-            for attachment in self.env['ir.attachment'].browse(kwargs.get('attachment_ids', [])):
-                self._create_invoice_from_attachment(attachment)
+            for attachment_id in self.env['ir.attachment'].search([('res_id', '=', self.id)]):
+                self._create_invoice_from_attachment(attachment_id)
```

Comment:
This is incorrect because `self.env['ir.attachment'].search([('res_id', '=', self.id)])` is not safe. Add at least the `res_model` to the domain.
Also, you absolutely need to retrieve the attachment from the `message_post` parameters. Making a search there doesn't make sense..